### PR TITLE
Wip/multilanguage pr

### PR DESCRIPTION
**Pull request by Mehrad0711**
Some changes from my side...

### CODE REVIEW

File: lib/i18n/default.js [hunk 1]
```diff
@@ -114,15 +114,19 @@ function isGoodWord(word) {
     // filter out words that cannot be in the dataset,
     // because they would be either tokenized/preprocessed out or
     // they are unlikely to be used with voice
-    return true;
+    return ! /[\n\r\t"]/.test(word);
```
Comment (gcampax) [inline lib/i18n/default.js:117]: Spaces (including \n, \r and the like) and quotes cannot be part of a word, so this test is redundant.

File: lib/i18n/japanese.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    'tt:stock_id': ['l.p.', 's.a.', 'plc', 'n.v', 's.a.b', 'c.v.'],
+    'org:freedesktop:app_id': ['gnome']
+};
+
+const ABBREVIATIONS = [
+    ['公司', '有限公司', '股份有限公司'],
+    ['高铁', '高速铁路'],
+    ['网路', '网络'],
+    ['&', '和'],
+];
```
Comment (gcampax) [inline lib/i18n/japanese.js:42]: How did you compile this abbreviation list?
Wikipedia for example lists KK as a standard abbreviation for a stock-traded company: https://en.wikipedia.org/wiki/Kabushiki_gaisha and GK for a limited-liability company: https://en.wikipedia.org/wiki/G%C5%8Dd%C5%8D_gaisha

File: lib/i18n/american-english.js [hunk 3]
```diff
@@ -327,7 +327,7 @@ function isGoodWord(word) {
     // filter out words that cannot be in the dataset,
     // because they would be either tokenized/preprocessed out or
     // they are unlikely to be used with voice
-    return /^([a-zA-Z0-9-][a-zA-Z0-9.-]*|'s|,|\?)$/.test(word);
+    return /^([a-zA-Z0-9-][a-zA-Z0-9.-]*|'s|,|\?|\.)$/.test(word);
```
[the remark this answers is not in the document]
Comment (Mehrad0711) [reply lib/i18n/american-english.js:341]: thanks for catching this; reverted. (it was a temporary fix for another problem)

File: lib/i18n/default.js (hunk 1 above)
Comment (Mehrad0711) [reply lib/i18n/default.js:117]: reverted.

File: lib/i18n/japanese.js (hunk 2 above)
Comment (Mehrad0711) [reply lib/i18n/japanese.js:42]: I did not compile this. It comes from Chinese package. I will omit it for now.
The language packages are very primitive right now and check basic things.
I will start writing the tokenizers for these languages which should help me learn new things to gradually improve the lang packages too.

**Pull request merged**